src/redisearch_rs/rlookup/src/lookup.rs

```diff
@@ -950,26 +952,31 @@ impl<'a> RLookup<'a> {
     /// nor in the schema.
     pub fn get_key_read(
         &mut self,
-        name: &'a CStr,
+        name: impl Into<Cow<'a, CStr>>,
         mut flags: RLookupKeyFlags,
     ) -> Option<&RLookupKey<'a>> {
         flags &= GET_KEY_FLAGS;
 
-        let available = self.keys.find_by_name(name).is_some();
+        let name = name.into();
+
+        let available = self.keys.find_by_name(&name).is_some();
         if available {
             // FIXME: Duplication because of borrow-checker false positive. Duplication means performance implications.
             // See <https://github.com/rust-lang/rust/issues/54663>
-            return self.keys.find_by_name(name).unwrap().into_current();
+            return self.keys.find_by_name(&name).unwrap().into_current();
```

Comment:
Is the comment above explaining why we can't use `if let Some(key) ...`? Can we?

Reply:
Yeah unfortunately this is the well-known "get-or-insert" borrow-checker false positive, where it (incorrectly) treats `self` as borrowed mutably for the entirety of the function body even though there is an early return.